remote/mdb_service_logging.go

```diff
@@ -5,18 +5,131 @@ import (
-func (s *mdbService) loggingGet(ctx context.Context, w io.Writer, msg mongowire.Message)         {}
-func (s *mdbService) loggingDelete(ctx context.Context, w io.Writer, msg mongowire.Message)      {}
-func (s *mdbService) loggingSendMessage(ctx context.Context, w io.Writer, msg mongowire.Message) {}
+func (s *mdbService) loggingSize(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	lc := s.serviceLoggingCacheRequest(ctx, w, msg, nil, LoggingCacheSizeCommand)
+	if lc == nil {
+		return
+	}
+
+	s.serviceLoggingCacheResponse(ctx, w, &loggingCacheSizeResponse{Size: lc.Len()}, LoggingCacheSizeCommand)
+}
+
+func (s *mdbService) loggingCreate(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	req := &loggingCacheCreateRequest{}
+	lc := s.serviceLoggingCacheRequest(ctx, w, msg, req, LoggingCacheCreateCommand)
+	if lc == nil {
+		return
+	}
+
+	cachedLogger, err := lc.Create(req.Params.ID, req.Params.Options)
+	if err != nil {
+		shell.WriteErrorResponse(ctx, w, mongowire.OP_REPLY, errors.Wrap(err, "could not create logger"), LoggingCacheCreateCommand)
+		return
+	}
+
+	s.serviceLoggingCacheResponse(ctx, w, makeLoggingCacheCreateAndGetResponse(cachedLogger), LoggingCacheCreateCommand)
+}
+
+func (s *mdbService) loggingGet(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	req := &loggingCacheGetRequest{}
+	lc := s.serviceLoggingCacheRequest(ctx, w, msg, req, LoggingCacheGetCommand)
+	if lc == nil {
+		return
+	}
+
+	cachedLogger := lc.Get(req.ID)
+	if cachedLogger == nil {
+		shell.WriteErrorResponse(ctx, w, mongowire.OP_REPLY, errors.New("named logger does not exist"), LoggingCacheGetCommand)
+		return
+	}
+
+	s.serviceLoggingCacheResponse(ctx, w, makeLoggingCacheCreateAndGetResponse(cachedLogger), LoggingCacheGetCommand)
+}
+
+func (s *mdbService) loggingDelete(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	req := &loggingCacheDeleteRequest{}
+	lc := s.serviceLoggingCacheRequest(ctx, w, msg, req, LoggingCacheDeleteCommand)
+	if lc == nil {
+		return
+	}
+
+	lc.Remove(req.ID)
+
+	s.serviceLoggingCacheResponse(ctx, w, nil, LoggingCacheDeleteCommand)
+}
+
+func (s *mdbService) loggingPrune(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	req := &loggingCachePruneRequest{}
+	lc := s.serviceLoggingCacheRequest(ctx, w, msg, req, LoggingCachePruneCommand)
+	if lc == nil {
+		return
+	}
+
+	lc.Prune(req.LastAccessed)
+
+	s.serviceLoggingCacheResponse(ctx, w, nil, LoggingCachePruneCommand)
+}
+
+func (s *mdbService) loggingSendMessage(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	req := &loggingSendMessageRequest{}
+	lc := s.serviceLoggingCacheRequest(ctx, w, msg, req, LoggingCacheDeleteCommand)
+	if lc == nil {
+		return
+	}
+
+	cachedLogger := lc.Get(req.Payload.LoggerID)
+	if cachedLogger == nil {
+		shell.WriteErrorResponse(ctx, w, mongowire.OP_REPLY, errors.New("named logger does not exist"), LoggingSendMessageCommand)
```

Comment:
Return if there's an error. The mongo shell doesn't support receiving multiple responses.